File: interview questions/longestPeak.ts
export function longestPeak(array: number[]) {
  let longestPeak: number = 0;
  let i: number = 0;

  while (i < array.length - 1) {
    const validPeak = array[i - 1] < array[i] && array[i] > array[i + 1];

    if (!validPeak) {
      i++;
      continue;
    }

    let leftIdx: number = i - 2;
    while (leftIdx >= 0 && array[leftIdx] < array[leftIdx + 1]) {
      leftIdx--;
    }

    let rightIdx: number = i + 2;
    while (rightIdx < array.length && array[rightIdx] < array[rightIdx - 1]) {
      rightIdx++;
    }

    const peakLength: number = rightIdx - leftIdx - 1;
    longestPeak = Math.max(peakLength, longestPeak);
    i = rightIdx;
  }

  return longestPeak;
}

File: interview questions/longestPeak.py
def longestPeak(array):
    longestPeak = 0
      i = 1

       while i < len(array)-1:
            left = array[i-1]
            curr = array[i]
            right = array[i+1]

            peakFound = left < curr and curr > right

            if not peakFound:
                i += 1
                continue

            leftIdx = i - 2
            while leftIdx >= 0 and array[leftIdx] < array[leftIdx+1]:
                leftIdx -= 1

            rightIdx = i + 2
            while rightIdx < len(array) and array[rightIdx] < array[rightIdx-1]:
                rightIdx += 1

            currPeakLength = rightIdx - leftIdx - 1
            longestPeak = max(currPeakLength, longestPeak)
            i = rightIdx

        return longestPeak
